src/Twilio/Security/RequestValidator.cs

```diff
@@ -124,23 +127,48 @@ private static bool SecureCompare(string a, string b)
             return mismatch == 0;
         }
 
-        private string RemovePort(UriBuilder uri)
+        private string RemovePort(string url)
         {
-            // UriBuilder.ToString() will not display the port 
-            // if the Port property is set to -1
+            var uri = new UriBuilder(url);
+            uri.Host = PreserveHostnameCasing(url);
             uri.Port = -1;
-            return uri.ToString();
+            var scheme = PreserveSchemeCasing(url);
+            return uri.Uri.OriginalString.Replace(uri.Scheme, scheme);
```

Comment:
Why is this doing a `Replace` instead of updating the `uri.Scheme`, like is done for `uri.host`?

Reply:
`uri.Scheme` does not let us override like `uri.Host` :(

It returns a lower cased value.